removed from models import* from bodhi.server.models.__init__

ankit01ojha opened the pull request:
removed the pacjkage bodhi.server.models and replaced it with bodhi.server.models.py.

ankit01ojha committed:
removes from models import* from bodhi.server.models.__init__ fixes #1057

ankit01ojha committed:
removed from models import* from bodhi.server.models.__init__ fixed #1057

ankit01ojha committed:
removed from models import* from bodhi.server.models.__init__ #fixed 1057

ankit01ojha committed:
removed from models import* from bodhi.server.models.__init__ #fixed 1057

ankit01ojha commented:
Sorry for four commits, I'am new to github :(

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -1,3 +1,145 @@
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+from bodhi.server.models import models
```
We don't really need this inside  `models.py`

trishnaguha commented:
Could you move all the import statements to the top? https://github.com/ankit01ojha/bodhi/blob/develop/bodhi/server/models.py#L159-L205

trishnaguha commented:
We don't need to import from `enum` anymore since it is inside `models.py` now https://github.com/ankit01ojha/bodhi/blob/develop/bodhi/server/models.py#L196

trishnaguha commented:
Please fix the branch conflict and rebase.

trishnaguha commented:
@ankit01ojha Did you run the test suite? If not can you please run the test suite? Because I think some of the tests are going to fail because of the import.

ankit01ojha commented:
Thanks @trishnaguha  i'll correct them.

ankit01ojha committed:
removed from models import* #fixed 1057

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -62,6 +46,167 @@
+
+import re
+
+from sqlalchemy.types import SchemaType, TypeDecorator, Enum
```
The idea is to keep all the import statements together. https://github.com/ankit01ojha/bodhi/blob/b7fdff2e6f66d8a487848f8424189bec798a55c9/bodhi/server/models.py#L1-L83
Please check the import guide https://www.python.org/dev/peps/pep-0008/#imports

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -62,6 +46,167 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```
Move the docstring before import statement https://www.python.org/dev/peps/pep-0008/#imports
Here is  duplication of docstring https://github.com/ankit01ojha/bodhi/blob/b7fdff2e6f66d8a487848f8424189bec798a55c9/bodhi/server/models.py#L49-L77

ankit01ojha committed:
removed from models import* #fixed 1057

removed frorm models import* fixed #1057

removed from models import*

removed from models import* from bodhi.server.models.__init__ fixed #1057

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -61,6 +60,135 @@
 except ImportError:
     log.warning("Could not import 'rpm'")
 
+import re
```
Importing this twice?

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -61,6 +60,135 @@
 
+import re
+
+from sqlalchemy.types import SchemaType, TypeDecorator, Enum
```
Could you move this to https://github.com/ankit01ojha/bodhi/blob/0ce799e298fa85d3f734debd4a5b439a50275a3a/bodhi/server/models.py#L45 ?

ankit01ojha replied:
Do I have to write this in place of https://github.com/ankit01ojha/bodhi/blob/0ce799e298fa85d3f734debd4a5b439a50275a3a/bodhi/server/models.py#L45 or before it?

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -14,6 +12,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```
There is still duplication here https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L173-L187

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -1,5 +1,3 @@
-# encoding: utf-8
```
Do not remove `encoding: utf-8` :).

trishnaguha commented:
Can you please fix the branch conflict? And squash and rebase your commits before pushing?

trishnaguha commented on .gitignore:
```diff
@@ -21,3 +21,5 @@ dogpile-cache.dbm*
 html-docs
 masher/
 Vagrantfile
+.idea/
```
It would be more perfect to have `*.idea/*` instead.

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -61,6 +61,133 @@
+        super(DeclEnumType, self).drop(bind, checkfirst)
+        t = self.dialect_impl(bind.dialect)
+        if t.impl.__class__ is not self.__class__ and isinstance(t, SchemaType):
+t.impl.drop(bind=bind, checkfirst=checkfirst)
```
Wrong indentation here.

trishnaguha commented on bodhi/server/models.py:
```diff
@@ -61,6 +61,133 @@
 except ImportError:
     log.warning("Could not import 'rpm'")
```
Don't we want to keep the first try block here? https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L28-L32

ankit01ojha replied:
Yes, we can.

trishnaguha replied:
This looks pretty messed up now. This is not what I meant https://github.com/ankit01ojha/bodhi/blob/b416dcbd62073fa3a495dcd9975c096504e92a0b/bodhi/server/models.py#L53-L60

ankit01ojha replied:
Should I undo it ?

ankit01ojha replied:
If I put the first try block here then can you please tell me where this try block should go:https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L59-L62   ?

trishnaguha replied:
We should avoid having `try-block` in the middle of the `import` statements.

ankit01ojha committed:
removed from models import* fixed #1057

ankit01ojha commented:
Sorry it again went two commits :(

ankit01ojha committed:
removed from models import* #fixed 1057

bowlofeggs commented:
@ankit01ojha This pull request has been sitting for a while. Do you still intend to work on it?

ankit01ojha commented:
@bowlofeggs sorry, for keeping this PR still for a while, I was having issues with my laptop that's why I was not able to work. Now its fixed , so i'll work.

ankit01ojha commented:
@bowlofeggs I have closed this PR as it looks pretty messed up :) 
A new PR has been opened https://github.com/fedora-infra/bodhi/pull/1190.

bowlofeggs commented:
On Fri, 2016-12-23 at 21:03 -0800, Ankit Raj Ojha wrote:
> @bowlofeggs I have closed this PR as it looks pretty messed up :)
> A new PR has been opened #1190.

Hello!

For next time, you can alter history in your own branch and force push
it to update the pull request. This will prevent you from having to
open a new one when you want to make changes.

ankit01ojha commented:
@bowlofeggs Thanks, I'll keep that for the next time. :)
